src/cli/cli.cpp

```diff
@@ -2626,29 +2626,34 @@ void Interpreter::ProcessNetworkDiagnostic(int argc, char *argv[])
 {
     ThreadError error = kThreadError_None;
     struct otIp6Address address;
-    uint8_t index = 2;
-    uint8_t tlvTypes[OT_NUM_NETDIAG_TLV_TYPES];
-    uint8_t count = 0;
+    uint8_t payload[2 + OT_NETWORK_DIAGNOSTIC_TYPELIST_MAX_ENTRIES];  // TypeList Type(1B), len(1B), type list
+    uint8_t payloadIndex = 0;
+    uint8_t paraIndex = 0;
```

Comment:
nit: rename to something like `paramIndex`?  Took me a while to realize this was short for "parameter index".